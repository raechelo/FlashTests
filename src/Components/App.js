import React, { Component } from 'react';
import questions from '../dataset.js';
import Board from './Board';
import Header from './Header';

class App extends Component {
  constructor() {
    super();
    this.state = {
      questions : []
    }
  }

  componentWillMount() {
    this.setState ( {questions: questions} )
  }
  
  render() {
    return (
      <div className="gameboard">
        <Header />
        <Board clues={this.state.questions}/>
      </div>
    );
  }
}

export default App;
