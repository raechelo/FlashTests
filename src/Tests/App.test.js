import React from 'react';
import App from '../Components/App';
import { shallow } from 'enzyme';

describe('App', () => {

  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state).toEqual( {questions: [], incorrect: [], correct: []} )
  });
})

