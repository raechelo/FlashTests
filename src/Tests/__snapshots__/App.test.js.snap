// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all the data passed in 1`] = `
<div
  className="gameboard"
>
  <Header />
  <Board
    iq={Array []}
    oldCards={[Function]}
    questions={Array []}
    saveToStore={[Function]}
  />
</div>
`;
